src/features/survey/survey-table.tsx

```diff
@@ -0,0 +1,90 @@
+import { useContext, useEffect } from "react"
+import { Link } from "react-router-dom"
+import { useAppDispatch } from "@hooks/useAppDispatch"
+import { useAppSelector } from "@hooks/useAppSelector"
+import { formatDateRange } from "@utils/format-date"
+import { Spinner } from "@components/ui/spinner/spinner"
+import { getByTemplateType, getByTemplateTypeSocket } from "@redux/slices/email-template-slice"
+import { WebSocketContext, type WebSocketType } from "@components/providers/websocket"
+import { Badge } from "@components/ui/badge/badge"
+import { getSurveyStatusVariant } from "@utils/variant"
+
+const mockSurveys = [
+  {
+    id: 1,
+    name: "Sample Survey 1",
+    status: "Ready",
+    startDate: "2024-03-01",
+    endDate: "2024-03-15",
+    description: "This is a sample survey for testing purposes.",
+    remarks: "Sample remarks",
+  },
+  {
+    id: 2,
+    name: "Sample Survey 2",
+    status: "Draft",
+    startDate: "2024-04-01",
+    endDate: "2024-04-15",
+    description: "Another sample survey for testing purposes.",
+    remarks: "Another sample remarks",
+  },
+  {
+    id: 3,
+    name: "Sample Survey 3",
+    status: "Ongoing",
+    startDate: "2024-04-01",
+    endDate: "2024-04-15",
+    description: "Another sample survey for testing purposes.",
+    remarks: "Another sample remarks",
+  },
+]
+
+export const SurveyTable = () => {
+  const appDispatch = useAppDispatch()
+  const { loading, surveys } = useAppSelector((state) => state.survey)
```

Comment:
use loading from `surveyAdministrations` state since we will be getting user survey administrations there for this list